Modules/IxNetRestApi.py

```diff
@@ -169,186 +175,86 @@ def __init__(self, apiServerIp=None, serverIpPort=None, serverOs='windows', linu
-                    self.httpHeader = self.sessionUrl.split('/ixnetworkweb')[0]
-
-            # Create new session: connectToLinuxApiServer API knows whether to create a new session or connect to an existing
-            #                     session by looking at the self.apiKey.
+            if username is None or password is None:
+                self.username = 'admin'
+                self.password = 'admin'
+
+            self.testPlatform = TestPlatform(ip_address=self.linuxApiServerIp,
+                                             rest_port=self.apiServerPort,
+                                             platform=self.serverOs,
+                                             verify_cert=verifySslCert,
+                                             log_file_name=self.restLogFile)
+            self.testPlatform.Authenticate(self.username, self.password)
+
+            if apiKey != None and sessionId == None:
+                raise IxNetRestApiException('Providing an apiKey must also provide a sessionId.')
+                # Connect to an existing session on the Linux API server
+            if apiKey and sessionId:
+                self.session = self.testPlatform.Sessions.find(Id=sessionId)
+                self.ixNetwork = self.session.Ixnetwork
+
+            if apiKey == None and sessionId:
+                self.session = self.testPlatform.Sessions.find(Id=sessionId)
+                self.ixNetwork = self.session.Ixnetwork
+
             if apiKey == None and sessionId == None:
-                self.connectToLinuxApiServer(self.linuxApiServerIp, self.apiServerPort, username=username, password=password,
-                                             verifySslCert=verifySslCert, timeout=self.linuxApiServerTimeout)
+                self.session = self.testPlatform.Sessions.add()
+                self.sessionId = self.session.Id
+                self.ixNetwork = self.session.Ixnetwork
+                ## TODO
+                # self.session_name = self.session.Name
 
             if licenseServerIp or licenseMode or licenseTier:
                 self.configLicenseServerDetails(licenseServerIp, licenseMode, licenseTier)
 
-        # For Linux API Server and Windoww Connection Mgr only: Delete the session when script is done if deleteSessionAfterTest = True.
-        self.deleteSessionAfterTest = deleteSessionAfterTest
-
-        match = re.match('http.*(/api.*/sessions/[0-9]+)/ixnetwork', self.sessionUrl)
-        self.headlessSessionId = match.group(1)
-
-        if includeDebugTraceback == False:
-            sys.tracebacklimit = 0
+            # For Linux API Server and Windoww Connection Mgr only: Delete the session when script is done
+            self.deleteSessionAfterTest = deleteSessionAfterTest
+            if includeDebugTraceback == False:
+                sys.tracebacklimit = 0
 
     def get(self, restApi, data={}, stream=False, silentMode=False, ignoreError=False, maxRetries=5):
```

Comment:
Why we need these functions if not used anymore? Also, you can add in the description it is not needed just added pass to avoid the exception.

Reply:
we'll change the description of the APIs, and keep the dummy API definition.